src/objects/Line.js

```diff
@@ -35,6 +35,75 @@ Line.prototype = Object.assign( Object.create( Object3D.prototype ), {
 
 	isLine: true,
 
+	computeLineDistances: ( function () {
+
+		var start = new Vector3();
+		var end = new Vector3();
+
+		return function computeLineDistances() {
+
+			var distance = 0;
+			var geometry = this.geometry;
+
+			if ( geometry.isBufferGeometry ) {
+
+				// we assume non-indexed geometry
+
+				if ( geometry.index === null ) {
+
+					var positionAttribute = geometry.attributes.position;
+					var lineDistances = [];
+
+					for ( var i = 0, l = positionAttribute.count; i < l; i ++ ) {
+
+						if ( i > 0 ) {
+
+							start.fromBufferAttribute( positionAttribute, i - 1 );
+							end.fromBufferAttribute( positionAttribute, i );
+
+							distance += start.distanceTo( end );
+
+						}
+
+						lineDistances[ i ] = distance;
+
+					}
+
+					geometry.addAttribute( 'lineDistance', new THREE.Float32BufferAttribute( lineDistances, 1 ) );
+
+				} else {
+
+					console.warn( 'THREE.Line.computeLineDistances(): Computation only possible with non-indexed BufferGeometry.' );
+
+				}
+
+			} else if ( geometry.isGeometry ) {
+
+				var vertices = geometry.vertices;
+
+				for ( var i = 0, l = vertices.length; i < l; i ++ ) {
+
+					if ( i > 0 ) {
+
+						start.copy( vertices[ i - 1 ] );
+						end.copy( vertices[ i ] );
+
+						distance += start.distanceTo( end );
+
+					}
+
+					geometry.lineDistances[ i ] = distance;
+
+				}
```

Comment:
```javascript

geometry.lineDistances[ 0 ] = 0;

for ( var i = 1, l = vertices.length; i < l; i ++ ) {

	geometry.lineDistances[ i ] += vertices[ i - 1 ].distanceTo( vertices[ i ] );
 
}
 ```

Reply:
That did not work for me since `geometry.lineDistances[ i ]` needs to be initialized with the previous value. Otherwise the summation does not work.
```javascript
geometry.lineDistances[ 0 ] = 0;

for ( var i = 1, l = vertices.length; i < l; i ++ ) {

        geometry.lineDistances[ i ] = geometry.lineDistances[ i - 1 ];
	geometry.lineDistances[ i ] += vertices[ i - 1 ].distanceTo( vertices[ i ] );
 
}
```